### PR TITLE
SISRP-17759-Ruby ~ backend support for SLR deeplink fetch

### PR DESCRIPTION
**Pull request by dfkaye**
https://jira.berkeley.edu/browse/SISRP-17759

This PR adds ruby infrastructure for fetching statement of legal residency deeplink ~ detail jira at https://jira.berkeley.edu/browse/SISRP-16994

### CODE REVIEW

**Comment by christianvuerings** [comment]
@dfkaye might be good to add an increased default cache time on this one.

**Comment by dfkaye** [comment]
@christianvuerings could you point me to an example?

**Comment by christianvuerings** [comment]
@dfkaye https://github.com/ets-berkeley-edu/calcentral/blob/db1d7b3953555ee6f95c7713b035901fd2f139b9/config/settings.yml#L275-L292

File: config/settings.yml [hunk 1]
```diff
@@ -288,6 +288,7 @@ cache:
     CampusSolutions::NameType: <%= 29.days %>
     CampusSolutions::SirConfig: <%= 24.hours %>
     CampusSolutions::State: <%= 29.days %>
+    CampusSolutions::StatementOfLegalResidency: <%= 29.days %>
```
Comment (redconfetti) [inline config/settings.yml]: @christianvuerings What cache time are you recommending for this?

Comment (christianvuerings) [inline config/settings.yml]: @redconfetti 24 hours might be a good start

Comment (dfkaye) [inline config/settings.yml]: @christianvuerings @redconfetti ~ OK, I will set it to 24 hours next push after standup

**Comment by dfkaye** [comment]
@christianvuerings do you think the names of the files/classes should be changed? maybe SLRDeeplink or similar?

**Comment by christianvuerings** [comment]
@dfkaye I'll leave that decision up to you.

**Comment by dfkaye** [comment]
@christianvuerings OK, I'll make that change, since it's a deeplink fetch

**Comment by dfkaye** [comment]
@christianvuerings @redconfetti I've renamed the files and both travis instances are passing ~ ready for another review...

**Pull request merged**